src/Translator/src/DependencyInjection/TranslatorCompilerPass.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symfony\UX\Translator\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Translation\TranslatorBagInterface;
+
+class TranslatorCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$this->hasValidTranslator($container)) {
+            $container->removeDefinition('ux.translator.cache_warmer.translations_cache_warmer');
+        }
+    }
+
+    private function hasValidTranslator(ContainerBuilder $containerBuilder): bool
```

Comment:
Do we need the method ?

Reply:
I personally feel that this reads better than something like:

```php
public function process(ContainerBuilder $container): void
{
    if (!$container->hasDefinition('translator') || i!s_subclass_of($container->getDefinition('translator')->getClass(), TranslatorBagInterface::class)) {
        $container->removeDefinition('ux.translator.cache_warmer.translations_cache_warmer');
    }
}
```

or the alternative

```php
public function process(ContainerBuilder $container): void
{
    if (!$container->hasDefinition('translator')) {
        $container->removeDefinition('ux.translator.cache_warmer.translations_cache_warmer');
        return;
    }
    $translator = $container->getDefinition('translator');
    if (i!s_subclass_of($translator->getClass(), TranslatorBagInterface::class)) {
        $container->removeDefinition('ux.translator.cache_warmer.translations_cache_warmer');
        return;
    }
}